test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -130,13 +130,29 @@ json_spirit::mValue BlockchainTestSuite::doTests(json_spirit::mValue const& _inp
         }
         else
         {
+            // Select test by name if --singletest is set and not filling state tests as blockchain
+            if (!Options::get().fillchain)
+            {
+                Options const& opt = test::Options::get();
+
+                // Select BC Test by singleTest
+                if (opt.singleTest && testname.find(opt.singleTestName) == string::npos)
+                    continue;
+
+                // Select BC Test by singleNet
+                if (!opt.singleTestNet.empty() && testname.rfind(opt.singleTestNet) == string::npos)
```

Comment:
There's no much difference between using `find` and `rfind` here if you do it like this (it will search the whole string anyway)
There are `boost::algorithm::starts_with`, `boost::algorithm::ends_with` for this

Reply:
So if you're worried, you should change it, it will help.

(`rfind("Byzantium")` will find it in your case)